devel/validate-platforms

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+#
+# Validate different platform builds of the final Nextstrain image.
+#
+set -euo pipefail
+
+# Set default values.
+registry=localhost:5000
+tag=latest
+
+# Read command-line arguments.
+while getopts "r:t:" opt; do
+    case "$opt" in
+        r) registry="$OPTARG";;
+        t) tag="$OPTARG";;
+        *) echo "Usage: $0 [-r <registry>] [-t <tag>]" 1>&2; exit 1;;
+    esac
+done
+
+IMAGE="$registry/nextstrain/base:$tag"
+PLATFORMS=(linux/amd64 linux/arm64)
+
+main() {
+    # Check software where downloaded versions are not deterministic at
+    # different points in time (e.g. if "latest" is downloaded).
+
+    local report_dir
+    report_dir="$(mktemp -dt "$(basename "$0")")"
+
+    for platform in "${PLATFORMS[@]}"; do
+        echo "[$platform] Pulling image..."
+        docker pull -q --platform "$platform" "$IMAGE"
+
+        echo "[$platform] Checking that the platform is expected..."
+        check-platform "$platform"
+
+        # Initialize a directory for the report file, ensuring slashes in the
+        # platform name are subdirs.
+        mkdir -p "$(dirname "$report_dir/$platform")"
+
+        # Create a report file for the platform.
+        echo "[$platform] Determining software versions..."
+        docker-run "$platform" bash -c '
+            PS4="\$ "
+            set -x
+
+            nextclade --version
+            augur --version
+            auspice --version
+            pip show evofr | grep Version
+            datasets --version
+            dataformat version
```

Comment:
_non-blocking_

Why report `datasets` and `dataformat`? I'd think we'd mostly (maybe even only) want to ensure that our first-party software is identical.

_[a couple minutes later]_

Ah, if the reason is because they're unpinned in our Dockerfile, then perhaps include a comment about this rationale here and there so we know a) why and b) to keep it synced?

Reply:
The comment here already existed, but maybe it could be worded more clearly?

https://github.com/nextstrain/docker-base/blob/468cb918971f161fce0f9df54c34770614a2f9ee/devel/validate-platforms#L24-L25

The newest force-push includes a comment in the Dockerfile:

https://github.com/nextstrain/docker-base/blob/c70aa6c2d37aed5572ae20f1fdfbd0c287559bb7/Dockerfile#L207-L211

Reply:
_non-blocking_

Yeah, this comment wasn't clear to me:

https://github.com/nextstrain/docker-base/blob/468cb918971f161fce0f9df54c34770614a2f9ee/devel/validate-platforms#L24-L25

I think I was thrown off by "latest", which in this context I was thinking about as an image tag. "Not deterministic at different points in time" is not super clear to me. (It's deterministic at different points in time, but time-varying. No?)

Maybe a reword to something like?

> Check that every platform image got the same versions of important (e.g. first-party) software where we don't pin a specific version (e.g. we install whatever the latest version is at build time).

Reply:
Oh yes, "time-varying" is what I meant. Reworded in [force-push](https://github.com/nextstrain/docker-base/compare/c70aa6c2d37aed5572ae20f1fdfbd0c287559bb7..7d8e9480ba91af55f0a225718b405f63570a465e).